src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -157,11 +161,69 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
             $condition = $globals['condition'];
         }
 
-        $route = $this->createRoute($globals['path'].$annot->getPath(), $defaults, $requirements, $options, $host, $schemes, $methods, $condition);
+        $path = $annot->getPath();
+        $locales = $annot->getLocales();
+        $hasLocalizedPrefix = !empty($globals['locales']);
+        $hasPrefix = $hasLocalizedPrefix || null !== $globals['path'];
+        $isLocalized = !empty($locales);
+        $hasPathOrLocales = null !== $path || $isLocalized;
+
+        if (!$hasPrefix && !$hasPathOrLocales) {
+            throw new \LogicException(sprintf('Annotation for "%s::%s" has no path.', $class->name, $method->name));
+        }
+
+        $paths = array();
+
+        if (!$hasPathOrLocales) {
+            if ($hasLocalizedPrefix) {
+                $paths = $globals['locales'];
+            } else {
+                $paths[] = $globals['path'];
+            }
+        } elseif (!$hasPrefix) {
+            if ($isLocalized) {
+                $paths = $locales;
+            } else {
+                $paths[] = $path;
+            }
+        } elseif ($hasLocalizedPrefix) {
+            if ($isLocalized) {
+                if ($missing = array_diff_key($globals['locales'], $locales)) {
+                    throw new \LogicException(sprintf('Route to "%s" is missing routes for locale(s) "%s', $class->name.'::'.$method->name, implode('", "', array_keys($missing))));
```

Comment:
Missing double quote and dot at end of message, same below.
All good on my side after that, thanks!

Reply:
Pushing the fix now.